src/initRules.js

```diff
@@ -175,17 +171,34 @@ module.exports = function(
           var timeoutId;
 
           return new Promise(function(resolve, reject) {
-            timeoutId = setTimeout(function() {
-              reject(
-                'A timeout occurred because the action took longer than ' +
-                  PROMISE_TIMEOUT / 1000 +
-                  ' seconds to complete. '
-              );
-            }, PROMISE_TIMEOUT);
-
-            Promise.resolve(
-              executeDelegateModule(action, syntheticEvent, [syntheticEvent])
-            ).then(resolve, reject);
+            var promiseTimeout = action.settings.timeout;
+
+            if (
+              promiseTimeout &&
+              (isNaN(promiseTimeout) || promiseTimeout < 0)
```

Comment:
When would promiseTimeout be NaN or less than 0? Is this attempting to catch a case where Blacksmith or Forge aren't working as expected?

Reply:
I am trying to catch the errors that probably will come from sandbox or rule editor if they aren't updated. Also, the developer can manually change the container.js file inside the sandbox. I was thinking to be more strict so the errors will be caught more easily.

Reply:
Since you commented about these errors, I can remove them if you think it's better.

Reply:
Turbine should be able to rely on a container that matches the container schema (https://github.com/adobe/reactor-turbine-schemas/blob/master/schemas/container.json#L58). The container build process inside the sandbox should ensure the container that is emitted matches the schema and the build process should fail otherwise.

Reply:
Not clear enough. Do you want the errors removed or not?

Reply:
Inside the JSON schema the timeout is not required. I don't think you can rely only on the schema.

Reply:
I've removed the code and the tests related timeout.

Reply:
Correct, the JSON schema says `timeout` is optional. That is fine, because it really is optional and no error should be thrown. The schema also says that `timeout` is a number that's a minimum of `0`. Therefore, there's no need for this error.